transcendence-app/src/auth/oauth42.guard.ts

```diff
@@ -1,22 +1,28 @@
 import {
-  BadGatewayException,
   ExecutionContext,
   Injectable,
+  Logger,
+  ServiceUnavailableException,
 } from '@nestjs/common';
 import { AuthGuard } from '@nestjs/passport';
 
 @Injectable()
 export class OAuth42Guard extends AuthGuard('oauth42') {
+  private readonly logger = new Logger(OAuth42Guard.name);
+
   async canActivate(context: ExecutionContext) {
     const result = (await super.canActivate(context)) as boolean;
     const request = context.switchToHttp().getRequest();
     await super.logIn(request);
     return result;
   }
 
-  handleRequest(err: any, user: any) {
+  handleRequest(err: Error, user: any) {
```

Comment:
User can be any too?

Reply:
I think it could be `any`, yes.

The definition of the interface:
```
handleRequest<TUser = any>(err: any, user: any, info: any, context: ExecutionContext, status?: any): TUser;
```

I've changed it to `handleRequest<User>(err: Error, user: User)`, and I don't know if it's doing anything for the typing system. We could leave it as documentation.